tasks/kernel_matrix_testing/selftest.py

```diff
@@ -47,40 +52,50 @@ def selftest_platforms_json(ctx: Context, _: bool) -> SelftestResult:
     return True, "platforms.json file exists, is readable and is correct"
 
 
-def selftest_prepare(ctx: Context, allow_infra_changes: bool, component: Component) -> SelftestResult:
-    """Ensures that we can run kmt.prepare for a given component.
+def selftest_prepare(ctx: Context, _: bool, component: Component, cross_compile: bool) -> SelftestResult:
+    """Ensures that we can run kmt.prepare for a given component."""
+    stack = f"selftest-prepare-{component}-xbuild{cross_compile}"
+    arch = Arch.local()
+    target = arch
+    if cross_compile:
+        if target == ARCH_AMD64:
+            target = ARCH_ARM64
+        else:
+            target = ARCH_AMD64
+        arch_arg = f"--arch={target.kmt_arch}"
+    else:
+        arch_arg = ""
 
-    If allow_infra_changes is true, the stack will be created if it doesn't exist.
-    """
-    stack = "selftest-prepare"
-    arch = full_arch("local")
-    vms = f"{arch}-debian11-distro"
+    vms = f"{target.name}-debian11-distro"
 
-    ctx.run(f"inv kmt.destroy-stack --stack={stack}", warn=True, hide=True)
+    ctx.run(f"inv kmt.destroy-stack --stack={stack}", warn=True)
     res = ctx.run(f"inv -e kmt.gen-config --stack={stack} --vms={vms} --init-stack --yes", warn=True)
     if res is None or not res.ok:
         return None, "Cannot generate config with inv kmt.gen-config"
 
-    if allow_infra_changes:
-        res = ctx.run(f"inv kmt.launch-stack --stack={stack}", warn=True)
-        if res is None or not res.ok:
-            return None, "Cannot create stack with inv kmt.create-stack"
-
-    compile_only = "--compile-only" if not allow_infra_changes else ""
-    res = ctx.run(f"inv -e kmt.prepare --stack={stack} --component={component} {compile_only}", warn=True)
+    res = ctx.run(f"inv -e kmt.prepare --stack={stack} --component={component} --compile-only {arch_arg}", warn=True)
     if res is None or not res.ok:
         return False, "Cannot run inv -e kmt.prepare"
 
-    paths = KMTPaths(stack, arch)
+    paths = KMTPaths(f"{stack}-ddvm", target)
     testpath = paths.secagent_tests if component == "security-agent" else paths.sysprobe_tests
     if not testpath.is_dir():
         return False, f"Tests directory {testpath} not found"
 
-    bytecode_dir = testpath / "pkg" / "ebpf" / "bytecode" / "build"
+    bytecode_dir = testpath / get_ebpf_build_dir(target)
     object_files = list(bytecode_dir.glob("*.o"))
     if len(object_files) == 0:
         return False, f"No object files found in {bytecode_dir}"
 
+    runtime_dir = bytecode_dir / "runtime"
+    runtime_files = list(runtime_dir.glob("*.c"))
+    if len(runtime_files) == 0:
```

Comment:
nitpick: it's sad to consume the generator this way. You could use a [peek](https://stackoverflow.com/a/664239), and we might have such method in `utils` (but it's a really small improvement :))